hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -315,7 +312,7 @@ public Builder withPartition(String partitionName) {
     public HoodieMergedLogRecordScanner build() {
       return new HoodieMergedLogRecordScanner(fs, basePath, logFilePaths, readerSchema,
           latestInstantTime, maxMemorySizeInBytes, readBlocksLazily, reverseReader,
-          bufferSize, spillableMapBasePath, instantRange, autoScan,
+          bufferSize, spillableMapBasePath, instantRange,
           diskMapType, isBitCaskDiskMapCompressionEnabled, withOperationField, true,
```

Comment:
looks like we are hardcoding force full scan to true here. guess autoScan above should be replaced w/ fullScan.

Reply:
So what's `forceFullScan` was `enableFullScan` before and it was always hard-coded to true.

Here, i'm simplifying the configuration as it's getting out of hand (there was `autoScan`, `enableFullScan` etc which actually were doing the same thing but on different levels) and keep just the `forceFullScan`